cmd/private-org-sync/main.go

```diff
@@ -460,35 +460,73 @@ func main() {
 		git:                  gitExec,
 		failOnNonexistentDst: o.failOnNonexistentDst,
 	}
+	var errs []error
+	var locations []location
+
+	whitelist := o.WhitelistOptions.WhitelistConfig.Whitelist
+	if len(whitelist) > 0 {
+		whitelistedLocations, err := getWhitelistedLocations(whitelist, syncer.git)
+		if err != nil {
+			errs = append(errs, err)
+		}
+		locations = append(locations, whitelistedLocations...)
+	}
 
-	var syncErrors []error
 	callback := func(_ *api.ReleaseBuildConfiguration, repoInfo *config.Info) error {
-		syncer.logger = config.LoggerForInfo(*repoInfo)
-		source := location{
+		locations = append(locations, location{
 			org:    repoInfo.Org,
 			repo:   repoInfo.Repo,
 			branch: repoInfo.Branch,
```

Comment:
Hmmm... I am wondering: if we accidentally happen to have a repo both in whitelist and having promoting config, we will end up having these locations in the worklist twice. Do we actually care? The locations will end up being synced twice, but that should not be much of an issue.

But if we want to avoid that, we could make the worklist a set instead of a list.

Reply:
Since there is not a good way to merge maps in golang, since the whole concept is like `c` and the map is already a pointer, I can just make the `getWhitelistedLocations` to fill the same map with here to avoid duplicates.